### DISCUSSION TITLE
Loop for each item from previous request

### DISCUSSION THREAD

**Discussion by QuAzI** [AnWeber/httpyac#552] [General]
I want to create a new request for each item returned by the previous request. By documentation, it should be something like 
```
> {%
  var items= [];
  response.body.forEach((element) => { items.push({
      "itemId": element["itemId"],
      "command": "...",
  })});
  client.global.set("items", items);
%}

# @loop for item of items
POST {{host}}/something/new
Content-Type: application/json

{{ item }}
```

But it doesn't work now because of AnWeber/vscode-httpyac/issues/227
And at the same time looks like JB ijhttp absolutely doesn't support `@loop`
So, how can I create some httpRequest for each item in other way to save compatibility with JB HTTP Client? Can I create some HttpRequests right in the first request script?

**Comment by AnWeber** [comment]
Neither JetBrains/ vsocde-restclient or Visual Studio does not support `@loop` statements (all uses some kind of http files).  This is a feature that is only supported in my extension. I also don't know how you get this use case into Intellij. Maybe with postscript which updates the variables and you need to execute for every loop step.
You should create a issue on youtrack, how to solve your problem

=> https://youtrack.jetbrains.com/issue/IDEA-320379/HTTP-Client-support-iterating-through-a-variable-collection

If Jetbrains supports a loop statement. I will implement the same api.

same feature in vscode-restclient: https://github.com/Huachao/vscode-restclient/issues/313
